projects/storefrontlib/src/cms-components/configurator/commons/config-add-to-cart-button/config-add-to-cart-button.component.html

```diff
@@ -1,12 +1,40 @@
 <ng-container *ngIf="configuration$ | async as configuration">
-  <div class="cx-config-add-to-cart-btn">
-    <div class="col-sm-6 col-md-4">
-      <button
-        class="cx-btn btn btn-block btn-primary"
-        (click)="onAddToCart(configuration.productCode, configuration.configId)"
-      >
-        {{ 'configurator.addToCart.button' | cxTranslate }}
-      </button>
+  <ng-container *ngIf="configuratorType$ | async as configuratorType">
+    <div class="cx-config-add-to-cart-btn">
+      <div class="col-sm-6 col-md-4">
+        <ng-container *ngIf="hasBeenAddedToCart$ | async as hasBeenAddedToCart">
+          <ng-container *ngIf="hasBeenAddedToCart.hasBeenAdded">
+            <button
+              class="cx-btn btn btn-block btn-primary"
+              (click)="
+                onAddToCart(
+                  configuration.productCode,
+                  configuration.configId,
+                  configuration.owner.key,
+                  configuratorType
+                )
+              "
+            >
+              {{ 'configurator.addToCart.buttonAfterAddToCart' | cxTranslate }}
+            </button></ng-container
+          >
+          <ng-container *ngIf="!hasBeenAddedToCart.hasBeenAdded">
```

Comment:
Could we do the ngIf only inside a button, that we don't have the method call (which is identical) twice?

<ng-container *ngIf="hasBeenAddedToCart.hasBeenAdded">
{{ 'configurator.addToCart.buttonAfterAddToCart' | cxTranslate }}
</ng-container
<ng-container *ngIf="!hasBeenAddedToCart.hasBeenAdded">
{{ 'configurator.addToCart.button' | cxTranslate }}
</ng-container

Reply:
Done...Didn't know that one can have this single line as part of an ng container